### PR TITLE
ARROW-2975: [Plasma] Fix TensorFlow operator compilation with pip package

### PR DESCRIPTION
**Pull request by pcmoritz**
This uses pyarrow.get_include() and pyarrow.get_library_dirs() instead of pkg-config to build the TensorFlow operator. The latter is not available in pip packages.

### CODE REVIEW

**Comment by pcmoritz** [comment]
I need a little help with the build system here. It works in normal operation, but not in the tests, since the arrow libraries are not available in the directories in `pyarrow.get_library_dirs()`. I created a JIRA here: https://issues.apache.org/jira/browse/ARROW-2976

**Comment by wesm** [comment]
Hm, does this impact the RC that is out? Can you comment on the vote thread?

File: python/pyarrow/tensorflow/build.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ NDEBUG="-DNDEBUG"
 
 g++ -std=c++11 -g -shared $PYARROW_TENSORFLOW_DIR/plasma_op.cc -o $PYARROW_TENSORFLOW_DIR/plasma_op.so \
     ${NDEBUG} \
-    `pkg-config --cflags --libs plasma arrow arrow-python` \
+    -I$1 $2 -lplasma -larrow_python -larrow \
```
Comment (cpcloud) [inline python/pyarrow/tensorflow/build.sh]: Quote these variables

**Comment by pcmoritz** [comment]
rebased on #2398 now

**Pull request closed without merging**